### ISSUE TITLE
A bunch of bugs with specific syntax raising errors during dts generation (Though they don't fail the build 🤷‍♂️ )

### ISSUE THREAD

**Issue by segevfiner** [qmhc/vite-plugin-dts#167]
### Describe the bug

The following trigger errors during dts generation:
1.
```ts
defineProps<{
    query: string | undefined;
}>()
```
2. [Using `v-bind` in CSS](https://vuejs.org/api/sfc-css-features.html#v-bind-in-css).
3. Compiling a plain JS Vue SFC using `script setup` and `defineProps`

<details><summary>The errors:</summary>

```
vite v4.1.1 building for production...
✓ 11 modules transformed.
dist/style.css                        0.09 kB │ gzip:  0.10 kB
dist/vite-vue-typescript-starter.js  50.36 kB │ gzip: 16.25 kB │ map: 519.07 kB

[vite:dts] Start generate declaration files...
src/components/Foo.vue.js:12:40 - error TS8016: Type assertion expressions can only be used in TypeScript files.

12 const __returned__ = { props: props as any }
                                          ~~~
src/components/HelloWorld.vue.ts:7:5 - error TS2769: No overload matches this call.
  The last overload gave the following error.
    Type '{ type: (StringConstructor | null)[]; required: true; }' is not assignable to type 'Prop<unknown, unknown> | null'.
      Types of property 'type' are incompatible.
        Type '(StringConstructor | null)[]' is not assignable to type 'true | PropType<unknown> | null | undefined'.
          Type '(StringConstructor | null)[]' is not assignable to type 'PropConstructor<unknown>[]'.
            Type 'StringConstructor | null' is not assignable to type 'PropConstructor<unknown>'.
              Type 'null' is not assignable to type 'PropConstructor<unknown>'.

7     msg: { type: [String, null], required: true }
      ~~~

  node_modules/@vue/runtime-core/dist/runtime-core.d.ts:666:25
    666 export declare function defineComponent<PropsOptions extends Readonly<ComponentPropsOptions>, RawBindings, D, C extends ComputedOptions = {}, M extends MethodOptions = {}, Mixin extends ComponentOptionsMixin = ComponentOptionsMixin, Extends extends ComponentOptionsMixin = ComponentOptionsMixin, E extends EmitsOptions = {}, EE extends string = string, I extends ComponentInjectOptions = {}, II extends string = string>(options: ComponentOptionsWithObjectProps<PropsOptions, RawBindings, D, C, M, Mixin, Extends, E, EE, I, II>): DefineComponent<PropsOptions, RawBindings, D, C, M, Mixin, Extends, E, EE>;
                                ~~~~~~~~~~~~~~~
    The last overload is declared here.
src/components/HelloWorld.vue.ts:13:3 - error TS2322: Type 'number' is not assignable to type 'string'.

13   "1-FOO": (FOO)
     ~~~~~~~

[vite:dts] Declaration files built in 624ms.

```
</details>

### Reproduction

https://stackblitz.com/edit/vitejs-vite-6b698p?file=src/components/HelloWorld.vue

### Steps to reproduce

1. Run `npm run build`

### System Info

```shell
System:
    OS: macOS 13.2
    CPU: (10) arm64 Apple M1 Pro
    Memory: 59.66 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.19.0 - ~/.nvm/versions/node/v16.19.0/bin/node
    npm: 9.4.1 - ~/.nvm/versions/node/v16.19.0/bin/npm
  npmPackages:
    @vitejs/plugin-vue: ^4.0.0 => 4.0.0 
    vite: ^4.0.0 => 4.0.4 
    vite-plugin-dts: 1.7.2 => 1.7.2
```

### Validations

- [X] Read the [FAQ](https://github.com/qmhc/vite-plugin-dts#faq).
- [X] Check that there isn't [already an issue](https://github.com/qmhc/vite-plugin-dts/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by segevfiner** [comment]
@qmhc Did you fix all the issues or only the CSS vars one? It might have auto closed this issue by accident.

**Comment by qmhc** [comment]
@segevfiner Yes, the two others fixed in #173.
